solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1604,6 +1609,73 @@ parameter_types! {
     pub const MaxLengthParaIds: u32 = 100u32;
     pub const MaxEncodedGenesisDataSize: u32 = 5_000_000u32; // 5MB
 }
+
+pub struct InnerStarlightRegistrar<AccountId, RegistrarManager, RegistrarWeightInfo>(
+    PhantomData<(AccountId, RegistrarManager, RegistrarWeightInfo)>,
+);
+impl<AccountId, RegistrarManager, RegistrarWeightInfo> RegistrarHandler<AccountId>
+    for InnerStarlightRegistrar<AccountId, RegistrarManager, RegistrarWeightInfo>
+where
+    RegistrarManager: RegistrarInterface<AccountId = AccountId>,
+    RegistrarWeightInfo: paras_registrar::WeightInfo,
+{
+    fn register(
+        who: AccountId,
+        id: ParaId,
+        genesis_storage: Vec<ContainerChainGenesisDataItem>,
+        head_data: Option<HeadData>,
+    ) -> DispatchResult {
+        // Return early if head_data is not specified
+        let genesis_head = match head_data {
+            Some(data) => data,
+            None => return Err(DispatchError::Other("HeadData not specified!")),
```

Comment:
Yes we can do it. I didn't take this approach initially because I wanted the `ContainerRegistrar` pallet to be as isolated as possible from any possible implementation of `InnerRegistrar`, but I'm also fine with adding them.

Applied in 1da160133ce626b398e73c7672930540e8662fa1.

Reply:
_fixed_